### DISCUSSION TITLE
Restrict updates to certificates matching a given CN

### DISCUSSION THREAD

**Discussion by sl8vz** [rauc/rauc#1256] [Q&A]
Hi,

We're using a 3rd party CA to sign the bundle and we would like to avoid having the intermediate certificate in the rootfs. Hence we want to rely on the chain on trust to validate the origin of the bundle and we would provide the intermediate certificate in the update bundle.

However by doing that any update signed by this CA could be accepted by rauc and I would like to know if there's a possibility to only accept a given CN

Regards,
Sebastien

**Comment by sl8vz** [comment]
Hi,
Any thoughts on this ? Am I missing something that could allow to match this requirement ?

**Comment by jluebbe** [comment]
Currently, there is no support for requiring a given CN.

You can use the [`allow-partial-chain` option](https://rauc.readthedocs.io/en/stable/reference.html#allow-partial-chain) in the `system.conf` and add only your intermediate certificate to the keyring. Then, other sub-trees of the 3rd party CA wouldn't be accepted as valid.

In almost all set-ups we've seen so far, I was easier to set up a separate (small) CA for signing RAUC bundles, as any pre-existing CAs and their use was not related to update signing in any way. Why do you prefer using a 3rd party CA instead of a separate CA?

**Reply by sl8vz** [reply]
We have chosen to use a 3rd party CA to avoid taking the responsibility of the involved security which is not our core business. The RAUC bundle is resigned using the 3rd party CA code signing facility.  
It's a common choice not to handle the CA internally.  
Verifying the CN would avoid having any intermediate certificate in the keyring and allows relying only on the system root certificates of public CA (AWS, SSL, Digicert, etc...)

**Reply by sl8vz** [reply]
@kavithabk is the root CA available in your system ?

**Reply by kavithabk** [reply]
@sl8vz yes the Root CA is available

**Reply by kavithabk** [reply]
@sl8vz @jluebbe ROOT CA was the issue , i am able to sign with external PKI now thanks for all the help
